frontend/src/queries/nodes/DataVisualization/Components/ConditionalFormatting/ConditionalFormattingTab.tsx

```diff
@@ -0,0 +1,216 @@
+import { conditionalFormattingLogic } from './conditionalFormattingLogic'
+
+const getRuleHeader = (rule: ConditionalFormattingRule): string => {
+    if (!rule.columnName || !rule.input) {
+        return 'New rule'
+    }
+
+    const template = FORMATTING_TEMPLATES.find((n) => n.id === rule.templateId)
+
+    if (!template) {
+        return 'New rule'
+    }
+
+    if (template.hideInput) {
+        return `${rule.columnName} ${template.label.toLowerCase()}`
+    }
+
+    return `${rule.columnName} ${template.label.toLowerCase()} ${rule.input}`
+}
+
+export const ConditionalFormattingTab = (): JSX.Element => {
+    const { isDarkModeOn } = useValues(themeLogic)
+    const { conditionalFormattingRules, conditionalFormattingRulesPanelActiveKeys } = useValues(dataVisualizationLogic)
+    const { addConditionalFormattingRule, setConditionalFormattingRulesPanelActiveKeys } =
+        useActions(dataVisualizationLogic)
+
+    return (
+        <div className="flex flex-col w-full ConditionalFormattingTab">
+            <p>You can add rules to make the cells in the table change color if they meet certain conditions.</p>
+
+            {conditionalFormattingRules.length > 0 && (
+                <LemonCollapse
+                    activeKeys={conditionalFormattingRulesPanelActiveKeys}
+                    onChange={(activeKeys) => setConditionalFormattingRulesPanelActiveKeys(activeKeys)}
+                    multiple={true}
+                    className="mb-2"
+                    size="small"
+                    panels={conditionalFormattingRules.map((rule) => ({
+                        key: rule.id,
+                        header: (
+                            <>
+                                <SeriesGlyph
+                                    style={{
+                                        borderColor: rule.color,
+                                        color: rule.color,
+                                        backgroundColor: isDarkModeOn
+                                            ? RGBToRGBA(lightenDarkenColor(rule.color, -20), 0.3)
+                                            : hexToRGBA(rule.color, 0.2),
+                                    }}
+                                >
+                                    <></>
+                                </SeriesGlyph>
+                                <span className="ml-2">{getRuleHeader(rule)}</span>
+                            </>
+                        ),
+                        content: <RuleItem rule={rule} key={rule.id} />,
+                    }))}
+                />
+            )}
+
+            <LemonButton
+                className="mt-1"
+                onClick={() => addConditionalFormattingRule()}
+                icon={<IconPlusSmall />}
+                fullWidth
+                type="secondary"
+            >
+                Add rule
+            </LemonButton>
+        </div>
+    )
+}
+
+const DEFAULT_PICKER_COLORS = [
```

Comment:
Where do these colors come from?

Reply:
Hand-picked by me from some online colour palettes - they're not perfect, but we don't have any _good_ defined colours to use that look good with a dark foreground colour (e.g., the colour of the text). Will chat to Cory at some point to define these better